### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by dhaiducek**

### CODE REVIEW

**Comment by mprahl** [comment]
@dhaiducek could you please use a max line length of say 100 characters? The VSCode "prettier" extension can automate this for you.

File: docs/policygenerator.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# About the PolicyGenerator plugin
+
+See the [PolicyGenerator reference YAML](./policygenerator-reference.yaml) to see the full structure of the PolicyGenerator manifest.
+
+## The Kustomize exec plugin explained
+
+The policy generator plugin is a binary written in Go. The binary accepts an input manifest of the `PolicyGenerator` kind in the `policy.open-cluster-management.io/v1` API. When Kustomize finds a manifest of that kind under the `generators:` array in `kustomization.yaml`, it looks in its directories for a binary of the same name ("PolicyGenerator") in the `policy.open-cluster-management.io/vi/policygenerator` directory and calls the binary as `<path>/PolicyGenerator <cached-manifest>` to run our generator as its plugin. The flag `--enable-alpha-plugins` must be specified in the kustomize command in order for our custom plugin to be enabled.
```
Comment (mprahl) [inline docs/policygenerator.md:7]:
```suggestion
The policy generator plugin is a binary written in Go. The binary accepts an input manifest of the `PolicyGenerator` kind in the `policy.open-cluster-management.io/v1` API. When Kustomize finds a manifest of that kind under the `generators:` array in `kustomization.yaml`, it looks in its directories for a binary of the same name ("PolicyGenerator") in the `policy.open-cluster-management.io/vi/policygenerator` directory and calls the binary as `<path>/PolicyGenerator <cached-manifest>` to run the generator as its plugin. The flag `--enable-alpha-plugins` must be specified in the Kustomize command in order for the external generator plugin to be enabled.
```

**Comment by dhaiducek** [comment]
Prettier apparently doesn't handle comments to rewrap. I had to install the Rewrap extension, but I think that'll be worth it for all the time it saved just this one time! 🙂

**Comment by mprahl** [comment]
> Prettier apparently doesn't handle comments to rewrap. I had to install the Rewrap extension, but I think that'll be worth it for all the time it saved just this one time!

Cool! Let me know when you push the change with it.

**Comment by dhaiducek** [comment]
It's already pushed (I had to rebase and force push because of DCO)

**Review by mprahl — APPROVED** [review]

**Pull request merged**